bindgen/ir/Struct.cpp

```diff
@@ -94,8 +90,8 @@ std::string Struct::str() const {
     return ss.str();
 }
 
-bool Struct::usesType(Type *type) const {
-    if (this == type) {
+bool Struct::usesType(std::shared_ptr<Type> type) const {
+    if (this == type.get()) {
```

Comment:
I'm not sure if this is the good way to check equality here.

Reply:
Right now it should be fine because parameter in `usesType` is always a `TypeDef`. There are no two different instances of the same `TypeDef` so it is okay to check pointers equality.
But in general `usesType` methods are not okay and should be fixed in future